#!/usr/bin/env python
PACKAGE = "usdev_framegrab"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("prompt",           str_t,   0, "Show some actions",  "Hello framegrabber!")
gen.add("update_index_list",bool_t,  1, "Update the index list of available video stream",  False)
gen.add("prev_video_index", bool_t,  1, "Previous video index",  False)
gen.add("next_video_index", bool_t,  1, "Next video index",  False)

gen.add("topleft_x0",       int_t,   2, "Cropping area x0",  0, 0, 1920)
gen.add("topleft_y0",       int_t,   3, "Cropping area y0",  0, 0, 1200)
gen.add("bottomright_x1",   int_t,   4, "Cropping area x1",  1280, 0, 1920)
gen.add("bottomright_y1",   int_t,   5, "Cropping area y1",  600, 0, 1200)

gen.add("auto_cropping",    bool_t,  6, "Automatically update the cropping area",  False)

gen.add("hflip_image",      bool_t,  7, "Horizontally flip the cropper image",  False)

gen.add("color_image",      bool_t,  8, "Return cropped color image",  False)

gen.add("save_settings",    bool_t,  9, "Save settings",  False)

exit(gen.generate(PACKAGE, "usdev_framegrab", "FGDynParam"))
